sdk/core/core-lro/samples-dev/customPoller.ts

```diff
@@ -0,0 +1,212 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * @summary demnostrates how to create a custom poller.
+ */
+
+import type { OperationState, OperationStatus, PollerLike } from "@azure/core-lro";
+import { delay } from "@azure/core-util";
+
+const DEFAULT_POLL_INTERVAL_IN_MS = 2000;
+
+type OperationConfig = { id: string };
```

Comment:
Should we add some doc comments (or just regular comments) explaining what this represents. IIUC, in this example, it simulates a service returning this contrived data, but in reality the data will contain additional fields as needed for polling is that right?

If I got it wrong, more of a reason to explain the purpose of these fields 😄

Reply:
Good point, addressed in https://github.com/Azure/azure-sdk-for-js/pull/32469/commits/9babca3eabce977777fdbb2aba1959735d48fcd2